File: 백준/Silver/1193. 분수찾기/분수찾기.py
X, i, sum = int(input()), 1, 0

while X > sum:
    sum += i
    i += 1

if i % 2:
    print(f'{X - sum + i - 1}/{sum - X + 1}')
else:
    print(f'{sum - X + 1}/{X - sum + i - 1}')

File: 백준/Silver/1193. 분수찾기/분수찾기.java
import java.util.*;

public class Main
{
	public static void main(String[] args) {
	    Scanner sc = new Scanner(System.in);
	    
	    int X = sc.nextInt(), i = 1, sum = 0;
	    
	    while (X > sum) {
            sum += i;
            i += 1;
        }
        
        if (i % 2 == 0) System.out.println(sum - X + 1 + "/" + (X - sum + i - 1));
        else System.out.print(X - sum + i - 1 + "/" + (sum - X + 1));
	}
}